frontend/src/components/Wallet.js

```diff
@@ -0,0 +1,184 @@
+		balance = utils.formatEther(data.balance);
+	}
+	
+	return <TableRow>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{data.pubAddress}
+		</TableCell>
+		<TableCell style={{ color: "white", borderBottom: "none" }}>
+			{balance}
+		</TableCell>
+	</TableRow>;
+}
+
+function EthereumAccountDetails ({ publicAddresses, setErrorNoMetamaskMessage }) {
+	const [detailData, setDetailData] = useState([]);
+
+	async function fetchAccountsDetails(publicAddresses) {
+		// TODO: publicAddresses is still not loaded. 
+		// Solve temporarily by waiting for 0.5 sec.
+		await delayMsec(500);
+		const provider = await detectEthereumProvider();
+		let details = [];
+		if (!provider) {
+			setErrorNoMetamaskMessage("Couldn't get data from Ethereum Provider. Please install MetaMask!");
+			console.error("no ethereum provider");
+			// Display only ETH address from backend
+			publicAddresses.map(pubAddress => {
+				details.push({ "pubAddress": pubAddress, "balance": NaN });
+			}) ;
+			setDetailData(details);
+			return;
+		}
+		const length = publicAddresses.length;
+		for (let i=0; i<length; i++) {
+			let pubAddress = publicAddresses[i];
+			const balance = await provider.request({ 
+				method: "eth_getBalance", 
+				params: [pubAddress, "latest"] 
+			});	// "Wei" balance in hex format, like 0x5a4e1804f198f1d99.
+			if (balance) {
+				details.push({ 
+					pubAddress: pubAddress,
+					balance: balance
+				});
+			}
+		}
+		setDetailData(details);
+	}
+
+	useEffect(() => {
+		fetchAccountsDetails(publicAddresses);
+	}, []);
+
+	return <TableContainer style={{ color: "white" }}>
+		<Table>
+			<TableBody>
+				<TableRow>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Account (Public Address)
+						</Typography>
+					</TableCell>
+					<TableCell style={{ color: "white", borderBottom: "none" }}>
+						<Typography variant="subtitle1" gutterBottom component="div">
+							Balance (Eth)
+						</Typography>
+					</TableCell>
+				</TableRow>
+				{detailData.map((value, key) => {
+					return <AccountEntry key={key} data={value} />;
+				})}
+			</TableBody>
+		</Table>
+	</TableContainer>;
+}
+
+/**
+ * Component to display and add Eth address from Wallet. Currently only Metamask is supported.
+ * @param user user object of the currently logged in user/admin
+ * @returns {JSX.Element}
+ */
+export default function Wallet({ user }) {
+	const [loadedAddresses, setLoadedAddresses] = useState(false);
+	const [publicAddresses, setPublicAddresses] = useState([]);
+	const [errorNoMetamaskMessage, setErrorNoMetamaskMessage] = useState("");
+
+	const history = useHistory();
+
+	const routeChange = (path) =>{
+		history.push(path);
+	};
+
+	const getEthAddressesFromServer = () => {
+		if (!loadedAddresses) {
+			appwriteApi.getOwnEthAddress(user.$id)
+				.then(result => {
+					// Set the address(es)
+					let currentPubAddr = publicAddresses;
```

Comment:
Spelling out this variable name would be at least as fine. Don't be afraid of long names. You can use autocompletion.

Spelling out `pubAddr` in line 157 would be even better.

Reply:
I would keep that. My eyes after hours got tired distinguishing `currentPublicAddress` and `publicAddresses`.

Reply:
You could have named it `currentWalletAddress`, easy to distinguish.